File: shapetrees-java-core/src/main/java/com/janeirodigital/shapetrees/core/methodhandlers/ValidatingMethodHandler.java
package com.janeirodigital.shapetrees.core.methodhandlers;

import com.janeirodigital.shapetrees.core.DocumentResponse;
import com.janeirodigital.shapetrees.core.exceptions.ShapeTreeException;
import com.janeirodigital.shapetrees.core.ShapeTreeRequest;

import java.util.Optional;

public interface ValidatingMethodHandler {
    Optional<DocumentResponse> validateRequest(ShapeTreeRequest shapeTreeRequest) throws ShapeTreeException;
}

File: asTypescript/packages/core/src/methodhandlers/ValidatingMethodHandler.ts
// Corresponding shapetrees-java package: com.janeirodigital.shapetrees.core.methodhandlers
import { DocumentResponse } from '../DocumentResponse';
import { ShapeTreeException } from '../exceptions/ShapeTreeException';
import { ShapeTreeRequest } from '../ShapeTreeRequest';
import * as Optional from 'java/util';

export interface ValidatingMethodHandler {

  validateRequest(shapeTreeRequest: ShapeTreeRequest): Optional<DocumentResponse> /* throws ShapeTreeException */;
}
